### PR TITLE
feat: overhaul jupyter notebook

### PR DESCRIPTION
**Pull request by gary-lloyd**
Remove JAP
Revert to Jupyter community containers
Note that no R libraries are installed but given the move to Conda these may not be required.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,126 +1,66 @@
...
 USER root
 
-EXPOSE 8888
-
-WORKDIR $HOME
-
-# Copy modified start script
-COPY ./start.sh /usr/local/bin/start.sh
+RUN apt-get update && apt-get install -yq --no-install-recommends \
+    vim \
```
Comment (fosj) [inline Dockerfile:50]: Is this necessary, looks like it may be left over from debugging?

File: Dockerfile [hunk 2]
```diff
@@ -1,126 +1,66 @@
...
 
-ENTRYPOINT ["tini", "--"]
-CMD ["start-notebook.sh"]
+# Add env-control executable to control creation/deletion of Conda Environments
+COPY env-control /usr/local/bin/env-control
+# Make env-control executable
+RUN chmod 755 /usr/local/bin/env-control
 
-USER $NB_USER
+USER $NB_UID
```
Comment (fosj) [inline Dockerfile]: Missing new line at end of file.

File: env-control [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+if [ "$1" == "add" ]; then
+  create_env
+  logger "Environment setup - $ENV_NAME complete, please restart browser for Kernel to be available."
+elif [ "$1" == "remove" ]; then
+  remove_env
+  logger "Environment setup - $ENV_NAME complete, please restart browser for Kernel to be available."
+else
+  print_help
+  exit 1
+fi
```
Comment (fosj) [inline env-control]: Missing new line at end of file.

File: env-control [hunk 4]
```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+# Wrapper to control creation/deletion of Conda Environments within JupyterLab hosted on Datalabs
```
Comment (fosj) [inline env-control:2]: This is pretty neat!

**Review by fosj — APPROVED** [review]
LGTM. Glad we can finally lose the old Jupyter images.

File: env-control (hunk 4 above)
Comment (alex-lee-tessella) [reply env-control:2]: I second that!

**Review by alex-lee-tessella — APPROVED** [review]

**Comment by fosj** [comment]
You might want to drop `start.sh`, as I don't think it is needed anymore.

File: env-control (hunk 4 above)
Comment (gary-lloyd) [reply env-control:2]: This is Iain's work that I copied from the Lab container

File: Dockerfile (hunk 1 above)
Comment (gary-lloyd) [reply Dockerfile:50]: I copied it from the lab file, will leave for now.

**Pull request merged**